REVAMPED: Auth, Payments, Refunds, Order

vinayak25 opened the pull request:
Contains following services:

1. Payment as a Service
2. Refunds as a Service
3. Orders as a Service
4. Authentication as a Service

To run the test case:
1. Clone the repo
2. Run "npm i"
3. Run "npm run test"

**For package dependencies, refer to package.json**

yashgoenka-im commented:
PASS  test/services/refunds.test.js
 PASS  test/services/auth.test.js
 PASS  test/utilities/nodeify.test.js
 PASS  test/utilities/form.test.js
 PASS  test/mojo.test.js
 PASS  test/services/payments.test.js (5.975s)
 FAIL  test/services/orders.test.js (8.732s)
  ● Test suite for orders service › CREATE: Order for payment request

    Request failed with status code 400

      at createError (node_modules/axios/lib/core/createError.js:16:15)
      at settle (node_modules/axios/lib/core/settle.js:17:12)
      at IncomingMessage.handleStreamEnd (node_modules/axios/lib/adapters/http.js:236:11)

  ● Test suite for orders service › GET: Transaction detail by id

    Request failed with status code 404

      at createError (node_modules/axios/lib/core/createError.js:16:15)
      at settle (node_modules/axios/lib/core/settle.js:17:12)
      at IncomingMessage.handleStreamEnd (node_modules/axios/lib/adapters/http.js:236:11)

Test Suites: 1 failed, 6 passed, 7 total
Tests:       2 failed, 20 passed, 22 total
Snapshots:   0 total
Time:        9.312s

3 test cases failed.

vinayak25 commented:
@yashgoenka-im I have fixed the test cases which were failing for orders suite. Also, found a vulnerability in orders service, callbacks were not getting passed for api calls, fixed that also
